media/plg_media-action_resize/js/resize.js

```diff
@@ -8,93 +14,93 @@ Joomla.MediaManager = Joomla.MediaManager || {};
-		Activate: function (mediaData) {
-			// Initialize
-			initResize(mediaData);
-		},
-		Deactivate: function () {
-		}
-	};
+  'use strict';
+
+  // Update image
+
+  var resize = function resize(width, height) {
+    // The image element
+    var image = document.getElementById('image-source');
+
+    // The canvas where we will resize the image
+    var canvas = document.createElement('canvas');
+    canvas.width = width;
+    canvas.height = height;
+    canvas.getContext('2d').drawImage(image, 0, 0, width, height);
+
+    // The format
+    var format = Joomla.MediaManager.Edit.original.extension === 'jpg' ? 'jpeg' : 'jpg';
+
+    // The quality
+    var quality = document.getElementById('jform_resize_quality').value;
+
+    // Creating the data from the canvas
+    Joomla.MediaManager.Edit.current.contents = canvas.toDataURL('image/' + format, quality);
+
+    // Updating the preview element
+    var preview = document.getElementById('image-preview');
+    preview.width = width;
+    preview.height = height;
+    preview.src = Joomla.MediaManager.Edit.current.contents;
+
+    // Update the width input box
+    document.getElementById('jform_resize_width').value = parseInt(width, 10);
+
+    // Update the height input box
+    document.getElementById('jform_resize_height').value = parseInt(height, 10);
+
+    // Notify the app that a change has been made
+    window.dispatchEvent(new Event('mediaManager.history.point'));
+  };
+
+  var initResize = function initResize() {
+    var funct = function funct() {
+      var image = document.getElementById('image-source');
+
+      var resizeWidthInputBox = document.getElementById('jform_resize_width');
+      var resizeHeightInputBox = document.getElementById('jform_resize_height');
+
+      // Update the input boxes
+      resizeWidthInputBox.value = image.width;
+      resizeHeightInputBox.value = image.height;
+
+      // The listeners
+      resizeWidthInputBox.addEventListener('change', function () {
+        resize(parseInt(_this.value, 10), parseInt(_this.value, 10) / (image.width / image.height));
+      });
+      resizeHeightInputBox.addEventListener('change', function () {
+        resize(parseInt(_this.value, 10) * (image.width / image.height), parseInt(_this.value, 10));
+      });
+
+      // Set the values for the range fields
+      var resizeWidth = document.getElementById('jform_resize_w');
+      var resizeHeight = document.getElementById('jform_resize_h');
+
+      resizeWidth.min = 0;
+      resizeWidth.max = image.width;
+      resizeWidth.value = image.width;
+
+      resizeHeight.min = 0;
+      resizeHeight.max = image.height;
+      resizeHeight.value = image.height;
+
+      // The listeners
+      resizeWidth.addEventListener('input', function resizeWidthF() {
+        resize(parseInt(this.value, 10), parseInt(this.value, 10) / (image.width / image.height));
+      });
+      resizeHeight.addEventListener('input', function resizeHeightF() {
```

Comment:
It is not possible to use an arrow function in line 88 and 91. If I use an arrow function, I can not resize the image. Does any one can explain me why I can not use an arrow function here?

Reply:
Arrow functions do not bind it’s own `this`. They lexically bind their context, so `this` refers to the originating context. The Value of `this` inside an arrow function is not determined by where it is used. Its determined by where the arrow function is defined. So in `this.value` will not refer to the value of the input element. We should get the value from the event target: `event.currentTarget.value`

Reply:
@dneukirchen  Thanks.